Formula/o/opentdf-cli.rb

```diff
@@ -0,0 +1,52 @@
+class OpentdfCli < Formula
+  desc "OpenTDF command-line interface"
+  homepage "https://docs.opentdf.io/"
+  url "https://github.com/opentdf/otdfctl.git",
+      tag:      "v0.11.4",
+      revision: "1e64065d08547658e436dc167b108c2a6e9c0ccd"
+  license "BSD-3-Clause-Clear"
+  head "https://github.com/opentdf/otdfctl.git", branch: "main"
+
+  livecheck do
+    url :stable
+    strategy :github_releases
+  end
+
+  keg_only :versioned_formula
+
+  # disable! date: "2025-01-01", because: :deprecated_upstream
+
+  depends_on "bash" => :build
+  depends_on "go" => :build
+
+  uses_from_macos "rsync" => :build
+
+  on_macos do
+    depends_on "coreutils" => :build
+  end
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X github.com/opentdf/otdfctl/pkg/config.Version=#{version}
+      -X github.com/opentdf/otdfctl/pkg/config.BuildTime=#{time.iso8601}
+      -X github.com/opentdf/otdfctl/pkg/config.CommitSha=#{Utils.git_head}
+    ].join(" ")
+    system "go", "build", *std_go_args(ldflags:), "-o", bin/"otdfctl"
+
+    generate_completions_from_executable(bin/"otdfctl", "completion", base_name: "otdfctl")
+
+    # TODO: install man docs
+    # Leave this step for the end as this dirties the git tree
+    # system "hack/update-generated-docs.sh"
+    # man1
+  end
+
+  test do
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
👍 There are a number of tests I can run which will result in a failure. I'll see if there is another test I can run, but I'll spend some time thinking if I have anything that I can test which doesn't result in failure.

I did follow kubectl as a template and modeled my tests after them

https://github.com/Homebrew/homebrew-core/blob/0200f867f8b67a91da18f2855ad338c59099ed52/Formula/k/kubernetes-cli.rb#L48-L55

I see that openshift-cli implements an additional test like you mention

https://github.com/Homebrew/homebrew-core/blob/0200f867f8b67a91da18f2855ad338c59099ed52/Formula/o/openshift-cli.rb#L61-L64

Reply:
Also for context, I followed this guide https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request which is the first entry on the list.
![CleanShot 2024-08-23 at 14 04 33@2x](https://github.com/user-attachments/assets/c50e5ff4-37d7-4311-99fb-27be9e20ded0)